# Generated by Django 4.2 on 2024-07-19 18:12

import django.core.validators
from django.db import migrations, models
import django.db.models.deletion
import django_jalali.db.models


class Migration(migrations.Migration):

    initial = True

    dependencies = [
        ('client_account', '0001_initial'),
    ]

    operations = [
        migrations.CreateModel(
            name='FinancialSourceModel',
            fields=[
                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('name', models.CharField(max_length=30)),
                ('type', models.CharField(choices=[('Card', 'Card'), ('Cash', 'Cash')], max_length=4)),
                ('card_number', models.CharField(blank=True, null=True, validators=[django.core.validators.RegexValidator(message='شماره کارت بایستی 16 رقم باشد.', regex='^\\d{16}$')])),
                ('is_calculate', models.BooleanField(default=True)),
                ('is_enable', models.BooleanField(default=True)),
                ('remain', models.PositiveBigIntegerField(default=0)),
                ('created_time', django_jalali.db.models.jDateTimeField(auto_now_add=True)),
                ('updated_time', django_jalali.db.models.jDateTimeField(auto_now=True)),
                ('account', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, related_name='financial_sources', to='client_account.accountmodel', verbose_name='account')),
            ],
            options={
                'verbose_name': 'Financial Source',
                'verbose_name_plural': 'Financial Sources',
                'db_table': 'financial source',
            },
        ),
        migrations.AddConstraint(
            model_name='financialsourcemodel',
            constraint=models.UniqueConstraint(fields=('account', 'name'), name='unique_account_name', violation_error_message='منبع مالی با این نام برای این جساب موجود است.'),
        ),
        migrations.AddConstraint(
            model_name='financialsourcemodel',
            constraint=models.UniqueConstraint(fields=('account', 'card_number'), name='unique_account_card_number', violation_error_message='منبع مالی با این شماره کارت برای این جساب موجود است.'),
        ),
    ]
